internal/provider/validators.go

```diff
@@ -95,8 +96,7 @@ func validateStringInSlice(valid []string, ignoreCase bool) schema.SchemaValidat
 // validateSemVer ensures a specified string is a SemVer.
 func validateSemVer(v interface{}, path cty.Path) diag.Diagnostics {
 	var diagnostics diag.Diagnostics
-
-	if !regexp.MustCompile(`^v?\d+.\d+.\d+$`).MatchString(v.(string)) {
+	if _, err := version.NewSemver(v.(string)); err != nil {
```

Comment:
This looks good to me. Given that go-version is used in the provider but not for the validation I want to confirm before merging that the consul resource is okay with a user specifying a pre-release version. In other words, is there a reason why the original implementor chose only to support official release versions?

Reply:
Disregard.

Reply:
> I want to confirm before merging that the consul resource is okay with a user specifying a pre-release version

Might have been a miss. But we do publicly release pre-release versions of our products (can be found [here](https://releases.hashicorp.com/consul/)). On the Consul side, we optionally enable testing out these pre-release versions on HCP for certain customers to unblock their testing/validation processes.